### PR TITLE
Refactoring AKS config to read from env Vars

### PR DESCRIPTION
**Pull request by adusumillipraveen**
Notes:
* Refactoring AKS cluster name and resource group to read from env Vars 
* Added default names dynamically for consistency.

### CODE REVIEW

File: vars/sectionCI.groovy [hunk 1]
```diff
@@ -27,19 +27,21 @@ def call(params) {
   def product = params.product
   def component = params.component
   def aksUrl
+  def environment = params.environment;
+  def parentEnvionment = params.parentEnvironment; //Used for secrets which are maintained only in parent environments.
```
Comment (timja) [inline vars/sectionCI.groovy]:
```suggestion
  def parentEnvironment = params.parentEnvironment; // Used for secrets which are maintained only in parent environments.
```

File: vars/withAksClient.groovy [hunk 2]
```diff
@@ -18,10 +16,14 @@ def withRegistrySecrets(Closure block) {
...
 
 
-def call(String subscription, Closure block) {
+def call(String subscription, String environment, Closure block) {
   withDocker('hmcts/cnp-aks-client:az-2.0.50-kubectl-1.12.2-helm-2.11', null) {
     withSubscription(subscription) {
       withRegistrySecrets {
+        def envName = environment.toUpperCase()
+        echo "envname : ${envName} "
+        env.AKS_CLUSTER_NAME = env."${envName}_AKS_CLUSTER_NAME" ?: "cnp-${environment}-cluster"
```
Comment (timja) [inline vars/withAksClient.groovy]: do we need the fallbacks here?
can't we just rely on the env variables always being set?

Comment (adusumillipraveen) [reply vars/withAksClient.groovy]: Ideally if we had good naming convention, we wouldn't need have to set them in environment. I would rather avoid the need of setting them  by following a convention.

File: vars/aksDeploy.groovy [hunk 3]
```diff
@@ -7,7 +7,7 @@ import uk.gov.hmcts.contino.Consul
 def call(DockerImage dockerImage, Map params) {
 
   def subscription = params.subscription
-  def environment = params.environment
+  def environment = params.parentEnvionment
```
Comment (luigibk) [inline vars/aksDeploy.groovy]: params.parentEnvironment

File: vars/sectionCI.groovy [hunk 4]
```diff
@@ -27,19 +27,21 @@ def call(params) {
...
+    withAksClient(subscription,environment) {
       def acr = new Acr(this, subscription, env.REGISTRY_NAME, env.REGISTRY_RESOURCE_GROUP)
       def dockerImage = new DockerImage(product, component, acr, new ProjectBranch(env.BRANCH_NAME).imageTag(), env.GIT_COMMIT)
 
       onPR {
         acr.retagForStage(DockerImage.DeploymentStage.PR, dockerImage)
 
         if (config.deployToAKS) {
-          withTeamSecrets(config, params.environment) {
+          withTeamSecrets(config, parentEnvironment) {
```
Comment (njrich28) [inline vars/sectionCI.groovy]: `withTeamSecrets` has built in support for overriding vaults between environments https://github.com/hmcts/cnp-jenkins-library/blob/master/vars/withTeamSecrets.groovy#L46.

The `config` object contains a map of environments (configurable by teams) to override the environment to use for vaults. The map defaults to empty `[:]` (i.e. no overrides) but we could default it to `['preview':'aat']`

File: vars/sectionCI.groovy [hunk 5]
```diff
@@ -27,19 +27,20 @@ def call(params) {
   def product = params.product
   def component = params.component
   def aksUrl
+  def environment = params.environment;
 
   Builder builder = pipelineType.builder
 
   if (config.dockerBuild) {
-    withAksClient(subscription) {
+    withAksClient(subscription,environment) {
```
Comment (timja) [inline vars/sectionCI.groovy]:
```suggestion
    withAksClient(subscription, environment) {
```

**Pull request merged**